cmd/clusterctl/client/init_test.go

```diff
@@ -159,6 +160,14 @@ func Test_clusterctlClient_InitImages(t *testing.T) {
 			wantErr:              true,
 			certManagerImagesErr: errors.New("failed to get cert images"),
 		},
+		{
```

Comment:
I took a closer look at this and turns out this test will pass even if we set `allowMissingProviderCRD` to `false`. This is becasue the test setup here uses fakeclient internally and fakeclient doent not care if a CRD is installed or not when doing operations like get or list. It might not be possible to test this just using fakeclient. 

So for now lets just drop this test (as it is not an accurate signal).

Reply:
Yes, If that is so then we should drop the tests.